rust/processor/src/processors/econia_processor.rs

```diff
@@ -118,21 +191,304 @@ fn insert_place_limit_order_events(
     Ok(())
 }
 
-fn insert_cancel_order_events(
+fn insert_place_market_order_events(
     conn: &mut PgConnection,
-    events: Vec<CancelOrderEvent>,
+    events: Vec<PlaceMarketOrderEvent>,
 ) -> Result<(), diesel::result::Error> {
     execute_with_better_error(
         conn,
         // See comment above re: conflicts
-        diesel::insert_into(cancel_order_events::table)
+        diesel::insert_into(place_market_order_events::table)
             .values(&events)
             .on_conflict_do_nothing(),
         None,
     )?;
     Ok(())
 }
 
+fn insert_place_swap_order_events(
+    conn: &mut PgConnection,
+    events: Vec<PlaceSwapOrderEvent>,
+) -> Result<(), diesel::result::Error> {
+    execute_with_better_error(
+        conn,
+        // See comment above re: conflicts
+        diesel::insert_into(place_swap_order_events::table)
+            .values(&events)
+            .on_conflict_do_nothing(),
+        None,
+    )?;
+    Ok(())
+}
+
+fn event_data_to_cancel_order_event(
+    event: &Event,
+    txn_version: BigDecimal,
+    event_idx: BigDecimal,
+    time: DateTime<Utc>,
+) -> anyhow::Result<CancelOrderEvent> {
+    let market_id = opt_value_to_big_decimal(event.data.get("market_id"))?;
+    let user = opt_value_to_string(event.data.get("user"))?;
+    let custodian_id = opt_value_to_big_decimal(event.data.get("custodian_id"))?;
+    let order_id = opt_value_to_big_decimal(event.data.get("order_id"))?;
+    let reason = opt_value_to_i16(event.data.get("reason"))?;
+
+    let cancel_order_event = CancelOrderEvent {
+        txn_version,
+        event_idx,
+        time,
+        user,
+        custodian_id,
+        order_id,
+        market_id,
+        reason,
+    };
+
+    Ok(cancel_order_event)
+}
+
+fn event_data_to_change_order_size_event(
```

Comment:
Were you able to test this (and the other events added)?
The trade simulation scripts are not generating some of these events.